### ISSUE TITLE
Rafcon freeze after global variable delete

### ISSUE THREAD

**Issue by franzlst** [DLR-RM/RAFCON#183]
Rafcon froze after I tried to delete a global variable.

For log details see the attached screenshot:
![rafcon_freeze](https://rmc-github-server/storage/user/23/files/12a4a8be-8eed-11e6-89ef-08ff920601ef)

Edit:
Found more information in the shell I started rafcon from:

```
13:23:34:    DEBUG - rafcon.statemachine.global_variable_manager:  Global variable explore_last_scan_position was deleted!
Traceback (most recent call last):
  File "/internal_path/common/packages/rafcon/0.7.11/lib/python2.7/rafcon/mvc/controllers/global_variable_manager.py", line 176, in on_name_changed
    self.model.global_variable_manager.set_variable(text, old_value, data_type=old_type)
  File "/internal_path/moro/packages/gtk_etc_python/0.0.9/sled11-x86-gcc4.x/lib/python2.7/site-packages/gtkmvc/observable.py", line 49, in wrapper
    res = _func(*args, **kwargs)
  File "/internal_path/common/packages/rafcon/0.7.11/lib/python2.7/rafcon/statemachine/global_variable_manager.py", line 51, in set_variable
    self.check_value_and_type(value, data_type)
  File "/internal_path/common/packages/rafcon/0.7.11/lib/python2.7/rafcon/statemachine/global_variable_manager.py", line 268, in check_value_and_type
    raise TypeError("Value '{0}' is not of data type '{1}'".format(value, data_type))
TypeError: Value '(-0.07926468116457917, 0.03690242487792794, 0.8194594119223388)' is not of data type 'None'
13:23:37:  WARNING - rafcon.mvc.controllers.global_variable_manager:  Delete of globale variable 'explore_last_scan_position' failed
```

Originally created by @peter-lehner  at 2016-10-10 11:27:09+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
@peter-lehner I have locked at the problem. I could not reproduce the freeze it-self but I found other simple problems. I think you are using the last release 0.7.11? I will talk to the developer who made the last change which are causing the problems I found. 

As far as I can see the newly insert lines never where tested with an human operating the GUI and the unit-test coverage for this widget is low.

As long as the failure is now in the release we will hurry to fix it and do a new release.


Originally created by @Rbelder  at 2016-10-11 09:30:28+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
This issue is solved by commit 87a13c3 and some following adjustments to the global variable manager. 

A bit off topic: I and @franzlst slightly changed the default behavior of the global variable manager to be more protective to already set data types and thereby maybe influence backward compatibility.

@sebastian-brunner Please check the following [lines](https://rmc-github-server/common/rafcon/commit/91145383d457853d0bd7bd69d259c7ff8059c9a9#diff-3a71fa89e73549bf8c9e28586119ddb2R108) in respective unit-test. I think we should leave it like it is but maybe we have to change it back.


Originally created by @Rbelder  at 2016-10-13 15:29:23+00:00 (moved from RMC internal repository)
